# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceSet.h
# Commit: ae329eaf22f7
# Full Hash: ae329eaf22f70199e5f3f9d16d4c3b7c50120732
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 4. Split FontFaceSetDocumentImpl from FontFaceSetImpl. r=emilio
#   
#   This patch splits document specific functionality from FontFaceSetImpl
#   into a new class FontFaceSetDocumentImpl. This will make it easier to
#   fork FontFaceSetImpl for workers.
# ==============================================================================

diff -r f724c21b1f15 -r ae329eaf22f7 layout/style/FontFaceSet.h
--- a/layout/style/FontFaceSet.h	Tue Jun 28 21:47:00 2022 +0000
+++ b/layout/style/FontFaceSet.h	Tue Jun 28 21:47:01 2022 +0000
@@ -16,7 +16,7 @@
 
 class nsFontFaceLoader;
 class nsIPrincipal;
-class nsPIDOMWindowInner;
+class nsIGlobalObject;
 struct RawServoFontFaceRule;
 
 namespace mozilla {
@@ -36,7 +36,8 @@
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(FontFaceSet, DOMEventTargetHelper)
 
-  FontFaceSet(nsPIDOMWindowInner* aWindow, dom::Document* aDocument);
+  static already_AddRefed<FontFaceSet> CreateForDocument(
+      dom::Document* aDocument);
 
   virtual JSObject* WrapObject(JSContext* aCx,
                                JS::Handle<JSObject*> aGivenProto) override;
@@ -112,6 +113,7 @@
  private:
   friend mozilla::dom::FontFaceSetIterator;  // needs GetFontFaceAt()
 
+  explicit FontFaceSet(nsIGlobalObject* aParent);
   ~FontFaceSet();
 
   /**